File: apps/flutter_app/lib/models/activity_log.dart
enum ActivityLogType {
  workspaceCreated,
  memberInvited,
  memberJoined,
  memberRemoved,
  listingSaved,
  listingRemoved,
  commentAdded,
  commentReplied,
  roleChanged,
  workspaceDeleted,
  workspaceUpdated,
  unknown
}

class ActivityLog {
  final String id;
  final String actorId;
  final ActivityLogType type;
  final String summary;
  final DateTime createdAt;
  final String? metadata;

  ActivityLog({
    required this.id,
    required this.actorId,
    required this.type,
    required this.summary,
    required this.createdAt,
    this.metadata,
  });

  factory ActivityLog.fromJson(Map<String, dynamic> json) {
    return ActivityLog(
      id: json['id'],
      actorId: json['actorId'],
      type: ActivityLogType.values.firstWhere(
        (e) => e.name.toLowerCase() == (json['type'] as String).toLowerCase(),
        orElse: () => ActivityLogType.unknown,
      ),
      summary: json['summary'],
      createdAt: DateTime.parse(json['createdAt']),
      metadata: json['metadata'],
    );
  }
}

File: backend/Valora.Domain/Entities/ActivityLog.cs
using Valora.Domain.Common;

namespace Valora.Domain.Entities;

public enum ActivityLogType
{
    WorkspaceCreated,
    MemberInvited,
    MemberJoined,
    MemberRemoved,
    ListingSaved,
    ListingRemoved,
    PropertySaved,
    PropertyRemoved,
    CommentAdded,
    CommentReplied,
    RoleChanged,
    WorkspaceDeleted,
    UserDeleted
}

public class ActivityLog : BaseEntity
{
    public Guid? WorkspaceId { get; set; }
    public Workspace? Workspace { get; set; }

    public required string ActorId { get; set; }
    public ApplicationUser? Actor { get; set; }

    public ActivityLogType Type { get; set; }

    public required string Summary { get; set; }
    public string? Metadata { get; set; }

    public Guid? TargetListingId { get; set; }
}
